app/components/QuestionForm/TypeForm.jsx

```diff
@@ -1,29 +1,53 @@
-import React, { memo, useCallback, useMemo } from 'react';
+import React, { memo, useCallback, useEffect, useMemo, useState } from 'react';
 import PropTypes from 'prop-types';
 import { Field } from 'redux-form/immutable';
 import { intlShape } from 'react-intl';
 
 import { FORM_TYPE } from './constants';
-
 import messages from './messages';
 import QuestionTypeField from './QuestionTypeField';
 import DescriptionList from '../DescriptionList';
 
-const TypeForm = ({ intl, change, locale, questionLoading, formValues }) => {
-  const onChange = useCallback(val => change(FORM_TYPE, val[0]), []);
+const TypeForm = ({
+  intl,
+  change,
+  locale,
+  questionLoading,
+  formValues,
+  isType,
+  setIsType,
+  isError,
+  setIsError,
+}) => {
+  const onChange = useCallback((val) => change(FORM_TYPE, val[0]), []);
+
+  const labelConditional = (n) => {
+    if (n === "1") return messages.generalQuestionDescriptionLabel.id;
+    if (n === "0") return messages.expertQuestionDescriptionLabel.id;
+    if (n === "2") return messages.tutorialQuestionDescriptionLabel.id;
+  };
+
+  const listConditional = (n) => {
+    if (n === "1") return messages.generalQuestionDescriptionList.id;
+    if (n === "0") return messages.expertQuestionDescriptionList.id;
+    if (n === "2") return messages.tutorialQuestionDescriptionList.id;
+  };
 
   const [descriptionListLabel, descriptionListItems] = useMemo(
     () => [
-      +formValues[FORM_TYPE]
-        ? messages.generalQuestionDescriptionLabel.id
-        : messages.expertQuestionDescriptionLabel.id,
-      +formValues[FORM_TYPE]
-        ? messages.generalQuestionDescriptionList.id
-        : messages.expertQuestionDescriptionList.id,
+      labelConditional(formValues[FORM_TYPE]),
+      listConditional(formValues[FORM_TYPE]),
     ],
-    [formValues],
+    [locale, formValues]
   );
 
+  useEffect(() => {
+    if (descriptionListLabel && descriptionListItems) {
+      setIsType(true);
+      setIsError(false);
+    }
+  }, [descriptionListLabel, descriptionListItems, isError]);
```

Comment:
isError is unnecessary in deps array, isn't it?

Reply:
Removed it